rpcserver.go

```diff
@@ -1000,6 +1000,14 @@ func (r *rpcServer) NewAddr(ctx context.Context,
 		}
 	}
 
+	// Check that the proof courier address is set. This should never
+	// happen, but we check anyway to avoid panics (possibly caused by
+	// future erroneous config changes).
+	if courierAddr == nil {
```

Comment:
I think this is already checked when creating the config `genServerConfig` but I guess it does not hurt to double check :+1: 

Also, couldn't we just eliminate the pointer from  `cfg.DefaultProofCourierAddr` itself?

Reply:
`cfg.DefaultProofCourierAddr` is optional because we also have a fallback address if a custom default isn't set.

Let me know if that's sub-optimal or if there's a better way.